### PR TITLE
Add lesson about asking for help

### PR DESCRIPTION
**Pull request by twalton83**
- [X] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).

#### 1.Describe the changes made:

Adds a lesson that aggregates our most commonly shared resources /sentiments towards asking questions. 

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number below:

#XXXXX

### CODE REVIEW

File: foundations/introduction/asking_for_help.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+Now, if you’re completely stuck with where to start, it’s completely fine to let people know that you’re stuck. Asking where you can start and what you can research to get on the right track empowers you to be able to resolve issues largely on your own in the future, and might even empower you to help others with the same issue later on. It is also recommended that you share your pseudo code so people can nudge you in the right direction or correct any misunderstandings you may have.
+
+3. Don’t take asking for more context to heart! People who volunteer in coding communities are here to help! But a question you may feel is coherent and “obvious” probably isn’t if you are being asked for more context. People take time out of their day to help, so make it as easy as possible for them to help you! There’s probably a good reason why they need more information, and trust their judgement and experience when they ask. Many people who help in coding chats are unpaid volunteers, and are in no way obligated to answer your query. But because they truly do want to help you, they will ask for more information when needed!
```
Comment (I3uckwheat) [inline foundations/introduction/asking_for_help.md:27]: You may want to mention here that while something may seem obvious to a beginner, it's sometimes nowhere near obvious to an expert. An expert knows about many, many more situations that could cause an issue someone is experiencing, and would want to refrain from sending a learner down the wrong path.

**Comment by I3uckwheat** [comment]
May also want to attach the help-vampire article

**Comment by twalton83** [comment]
@I3uckwheat Re: the vamp article, wasn’t sure if it was “too severe” for content that’s meant to be early on. Thoughts on this?

**Comment by I3uckwheat** [comment]
@twalton83 I don't think so, it's not accusing someone. It's just asking them to ensure they don't become one.

**Review by I3uckwheat — APPROVED** [review]

**Comment by leosoaivan** [comment]
:shipit:

**Pull request merged**